DigitalLearningSolutions.Data.Migrations/202204071530_PopulateUsersTableFromAccountsTables.cs

```diff
@@ -0,0 +1,415 @@
+                    10,
+                    NULL,
+                    ProfileImage_deprecated,
+                    Active,
+                    ResetPasswordID_deprecated,
+                    TCAgreed_deprecated,
+                    FailedLoginCount_deprecated,
+                    0,
+                    NULL,
+                    0,
+                    GETUTCDATE(),
+                    GETUTCDATE()
+                    FROM AdminAccounts"
+            );
+
+            // 3. Update AdminAccounts to reference Users.ID
+            connection.Execute(
+                @"UPDATE AdminAccounts
+                    SET
+                        UserID = (SELECT ID FROM Users WHERE Email = PrimaryEmail),
+                        Email = NULL"
+            );
+
+            // 4. DelegateAccount
+
+            // Transfer all delegates with unique emails not already in the Users table
+            connection.Execute(
+                @"INSERT INTO dbo.Users (
+                    PrimaryEmail,
+                    PasswordHash,
+                    FirstName,
+                    LastName,
+                    JobGroupID,
+                    ProfessionalRegistrationNumber,
+                    ProfileImage,
+                    Active,
+                    ResetPasswordID,
+                    TermsAgreed,
+                    FailedLoginCount,
+                    HasBeenPromptedForPrn,
+                    LearningHubAuthId,
+                    HasDismissedLhLoginWarning,
+                    EmailVerified,
+                    DetailsLastChecked
+                )
+                SELECT Email,
+                    COALESCE(OldPassword, ''),
+                    COALESCE(FirstName_deprecated, ''),
+                    COALESCE(LastName_deprecated, ''),
+                    JobGroupID_deprecated,
+                    ProfessionalRegistrationNumber_deprecated,
+                    ProfileImage_deprecated,
+                    Active,
+                    ResetPasswordID_deprecated,
+                    NULL,
+                    0,
+                    HasBeenPromptedForPrn_deprecated,
+                    LearningHubAuthID_deprecated,
+                    HasDismissedLhLoginWarning_deprecated,
+                    GETUTCDATE(),
+                    GETUTCDATE()
+                    FROM DelegateAccounts
+                    WHERE Email IN (
+                        SELECT Email FROM DelegateAccounts
+                        WHERE Email IS NOT NULL AND TRIM(Email) IS NOT NULL AND TRIM(Email) <> ''
+                        GROUP BY Email
+                        HAVING COUNT(*) = 1
+                        EXCEPT
+                        SELECT PrimaryEmail FROM Users)"
+            );
+            connection.Execute(
+                @"UPDATE DelegateAccounts
+                    SET
+                        UserID = (SELECT ID FROM Users WHERE Email = PrimaryEmail),
+                        Email = NULL,

```

Comment:
I've moved this to the end since Steve also suggested this https://github.com/TechnologyEnhancedLearning/DLSV2/pull/1065#discussion_r850187659 and we can do it all at once at the end